src/main/scala/org/broadinstitute/dsde/agora/server/webservice/QueryHandler.scala

```diff
@@ -59,8 +68,14 @@ class QueryHandler(authorizationProvider: AuthorizationProvider) extends Actor {
             entityTypes: Seq[AgoraEntityType.EntityType],
             username: String): Unit = {
     val entities = agoraBusiness.find(agoraSearch, agoraProjection, entityTypes, username)
-    authorizationProvider.filterByReadPermissions(authorizationProvider.authorizationsForEntities(entities, username))
-    context.parent ! RequestComplete(entities)
+    try {
+      val authorizedEntities = authorizationProvider.filterByReadPermissions(authorizationProvider.authorizationsForEntities(entities, username))
```

Comment:
making two subsequent calls to the auth provider seems a bit silly. Maybe we should create a few helper methods like
filterEntitiesByReadPermissions(agoraEntities: Seq[AgoraEntity], username: String)
then we can compose this line like
context.parent ! RequestComplete(filterEntitiesByReadPermissions(entities, username)